modules/circuits/controllers/NodesController.php

```diff
@@ -70,14 +73,138 @@ public function actionCreate(){ // this route manages the view and backend logic
     }
     
     public function actionShow() {   // this function manages the mapping of SDX-topology and displays on the MEICAN UI
-   
+      
+      $api_url=API_URL;
+      $meican_url=MEICAN_URL;
+
+      $userId = Yii::$app->user->id;
+      $rows = (new \yii\db\Query())
+          ->select(['token', 'expiration'])
+          ->from('meican_cilogon_auth')
+          ->where(['user_id' => $userId])
+          ->all();
+      
+      date_default_timezone_set('America/New_York');
```

Comment:
add comments regarding how the token is updating/inserting and how the expiration is working.

Reply:
Hi Usman, I have added the comments and pushed the code.